File: crates/vexus-index/tests/fixtures/cpp/sample.cpp
#include "cart.hpp"
#include <vector>

namespace shop {

// One shopping cart.
class Cart {
 public:
  int total(int tax) { return subtotal() + tax; }

 private:
  int subtotal() { return items_; }
  int items_ = 0;
};

struct Point {
  int x;
  int y;
};

enum class Color { Red, Green };

template <typename T>
T clamp_min(T value, T floor) {
  return value < floor ? floor : value;
}

int helper(int v);

}  // namespace shop

int shop::helper(int v) { return shop::clamp_min(v, 0); }

typedef int Cents;
using Money = int;

File: crates/vexus-index/tests/fixtures/ruby/sample.rb
require "json"
require_relative "helper"

module Shop
  # One shopping cart.
  class Cart
    def initialize(items)
      @items = items
    end

    # Total in cents.
    def total(tax)
      subtotal + tax
    end

    def subtotal
      @items.sum
    end

    def self.build(items)
      new(items)
    end
  end
end
